### PR TITLE
#160609559 enable user following

### PR DESCRIPTION
**Pull request by MuhanguziDavid**
### What does this PR do?

It allows users to follow and unfollow other users

### Description of tasks to be completed?

Currently, a user cannot follow or be followed by other users
This PR adds functionality to allow logged in users to follow and unfollow other users

### How should this be manually tested?

Make migrations `python manage.py makemigrations` and `python manage.py migrate`
Run the server `python manage.py runserver`

Open postman and:

***Follow a user***
POST `http://127.0.0.1:8000/api/profiles/<username>/follow/`

***Unfollow user***
DELETE `http://127.0.0.1:8000/api/profiles/<username>/unfollow/`

### What are the relevant pivotal tracker stories?

[#160609559](https://www.pivotaltracker.com/story/show/160609559)

### Any background context you want to add?

Please have at least two users registered before testing this feature.

### Screenshots
*Result of following a user*
follow

*Result of unfollowing a user*
unfollow

### CODE REVIEW

**Comment by RonKbS** [comment]
Return usernames instead of ids for the followers when a user accesses their profile

File: authors/apps/profiles/tests.py [hunk 1]
```diff
@@ -82,3 +119,57 @@ def test_unauthenticated_retrieval_of_all_user_profiles(self):
...
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+    
+    def test_follow_yourself(self):
+        response = self.client.delete(
+            '/api/profiles/{}/unfollow/'.format(self.username)
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_follow_non_esixtent_user(self):
```
Comment (RonKbS) [inline authors/apps/profiles/tests.py]: Correct the spelling of existent please.

**Comment by lym** [comment]
@MuhanguziDavid Please Update your PR title to be aligned with our project conventions.

File: authors/apps/profiles/tests.py (hunk 1 above)
Comment (MuhanguziDavid) [reply authors/apps/profiles/tests.py]: Thanks, @RonKbS, the spelling has been corrected

**Comment by lym** [comment]
LGTM

**Pull request merged**